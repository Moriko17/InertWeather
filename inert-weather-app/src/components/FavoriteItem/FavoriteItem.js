import React from "react";
import "../../styles/styles.css";
import Loading from "../Loading/Loading";
import Error from "../Error/Error";

class FavoriteItem extends React.Component {
  constructor(props) {
    super(props);
    this.props.fetchWeather(this.props.city.cityName, this.props.city.id);
  }
  render() {
    const { city } = this.props;
    return (
      <div className="favorite-container">
        <div className="favorite-bar">
          <p>{city.cityName}</p>
          {city.fetchedData ? <p>{city.fetchedData.temp} Â°C</p> : <p></p>}
          <button
            onClick={() => {
              this.props.deleteCityOnSrv(city.cityName);
              this.props.deleteCity(city.id);
            }}
          >
            X
          </button>
        </div>
        {city.isLoading === true || city.errorMessage !== "" ? (
          city.isLoading ? (
            <Loading />
          ) : (
            <Error errorMessage={city.errorMessage} />
          )
        ) : city.fetchedData ? (
          <div className="favorite-bot">
            <div className="favorite-q">
              <p>Wind:</p>
              <p>Clouds:</p>
              <p>Pressure:</p>
              <p>Humidity:</p>
              <p>Coord:</p>
            </div>
            <div className="favorite-a">
              <p>{city.fetchedData.wind} m/s</p>
              <p>{city.fetchedData.cloudState}</p>
              <p>{(city.fetchedData.pressure * 0.750062).toFixed(0)} mmHg</p>
              <p>{city.fetchedData.humidity} %</p>
              <p>
                [{city.fetchedData.coord.lat}, {city.fetchedData.coord.lon}]
              </p>
            </div>
          </div>
        ) : (
          false
        )}
      </div>
    );
  }
}

export default FavoriteItem;
