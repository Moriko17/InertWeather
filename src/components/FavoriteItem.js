import React from "react";
import "../styles/styles.css";

class FavoriteItem extends React.Component {
  constructor(props) {
    super(props);
    this.props.fetchWeather(this.props.city.cityName, this.props.city.id);
  }
  render() {
    const { city } = this.props;
    return (
      <div key={city.id}>
        <div>
          <div>
            <p>{city.cityName}</p>
            <p>{city.temp}</p>
            <button onClick={() => this.props.deleteCity(city.id)}>X</button>
          </div>
          {city.isLoading === true || city.errorMessage !== "" ? (
            city.isLoading ? (
              <p>Loading</p>
            ) : (
              <p>{city.errorMessage}</p>
            )
          ) : city.fetchedData ? (
            <div>
              <div className="favorite-q">
                <p>Wind:</p>
                <p>Clouds:</p>
                <p>Pressure:</p>
                <p>Humidity:</p>
                <p>Coord:</p>
              </div>
              <div className="favorite-a">
                <p>{city.fetchedData.wind} m/s</p>
                <p>{city.fetchedData.cloudState}</p>
                <p>{(city.fetchedData.pressure * 0.750062).toFixed(0)} mmHg</p>
                <p>{city.fetchedData.humidity} %</p>
                <p>
                  [{city.fetchedData.coord.lat}, {city.fetchedData.coord.lon}]
                </p>
              </div>
            </div>
          ) : (
            false
          )}
        </div>
      </div>
    );
  }
}

export default FavoriteItem;
