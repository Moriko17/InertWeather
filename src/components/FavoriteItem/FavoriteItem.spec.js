import React from "react";
import renderer from "react-test-renderer";
import FavoriteItem from "./FavoriteItem";

describe("AddToFavorite renders right", () => {
  const errorCity = {
    cityName: "12",
    errorMessage: 404,
    id: 55,
    isLoading: false
  };

  const filledCity = {
    cityName: "Marunouchi",
    erroeMessage: "",
    isLoading: false,
    fetchedData: {
      cityName: "Marunouchi",
      cloudState: "few clouds",
      coord: {
        lat: 35.68,
        lon: 139.76
      },
      humidity: 63,
      pressure: 1014,
      temp: 13.11,
      wind: 2.6
    },
    id: 60
  };

  const loadingCity = {
    cityName: "Marunouchi",
    erroeMessage: "",
    isLoading: true,
    id: 60
  };

  it("Renders correctly with error catched", () => {
    const tree = renderer
      .create(
        <FavoriteItem
          fetchWeather={() => {}}
          deleteCity={() => {}}
          city={errorCity}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renders correctly with fetched data", () => {
    const tree = renderer
      .create(
        <FavoriteItem
          fetchWeather={() => {}}
          deleteCity={() => {}}
          city={filledCity}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renders correctly while loading", () => {
    const tree = renderer
      .create(
        <FavoriteItem
          fetchWeather={() => {}}
          deleteCity={() => {}}
          city={loadingCity}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
