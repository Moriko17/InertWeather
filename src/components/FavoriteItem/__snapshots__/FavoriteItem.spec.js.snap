// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToFavorite renders right Renders correctly while loading 1`] = `
<div
  className="favorite-container"
>
  <div
    className="favorite-bar"
  >
    <p>
      Marunouchi
    </p>
    <p />
    <button
      onClick={[Function]}
    >
      X
    </button>
  </div>
  <div
    className="loading"
  >
    <div
      className="css-0"
    >
      <div
        className="css-110do72"
      />
      <div
        className="css-1by9rry"
      />
      <div
        className="css-110do72"
      />
    </div>
  </div>
</div>
`;

exports[`AddToFavorite renders right Renders correctly with error catched 1`] = `
<div
  className="favorite-container"
>
  <div
    className="favorite-bar"
  >
    <p>
      12
    </p>
    <p />
    <button
      onClick={[Function]}
    >
      X
    </button>
  </div>
  <div
    className="error"
  >
    <p>
      Something goes wrong: 
      404
    </p>
  </div>
</div>
`;

exports[`AddToFavorite renders right Renders correctly with fetched data 1`] = `
<div
  className="favorite-container"
>
  <div
    className="favorite-bar"
  >
    <p>
      Marunouchi
    </p>
    <p>
      13.11
       Â°C
    </p>
    <button
      onClick={[Function]}
    >
      X
    </button>
  </div>
  <div
    className="error"
  >
    <p>
      Something goes wrong: 
    </p>
  </div>
</div>
`;
